tournamentcontrol/competition/tests/test_logo_fields.py

```diff
@@ -0,0 +1,240 @@
+        with self.login(self.superuser):
+            self.assertGoodView(viewname, test_query_count=test_query_count, *args)
+            if kwargs:
+                self.post(viewname, *args, **kwargs)
+                self.response_302()
+    
+    def assertGoodDeleteView(self, viewname, *args):
+        self.assertLoginRequired(viewname, *args)
+        with self.login(self.superuser):
+            self.get(viewname, *args)
+            self.response_405()
+            self.post(viewname, *args)
+            self.response_302()
+    
+    def assertGoodNamespace(self, instance, **kwargs):
+        namespace = instance._get_admin_namespace()
+        args = instance._get_url_args()
+        self.assertGoodEditView("%s:add" % namespace, *args[:-1], **kwargs)
+        self.assertGoodEditView("%s:edit" % namespace, *args, **kwargs)
+        self.assertGoodDeleteView("%s:delete" % namespace, *args)
+
+
+class LogoFieldTestCase(TestCase):
+    """Test that logo fields accept valid file formats via admin views."""
+    
+    @classmethod
+    def setUpTestData(cls):
+        """Create test data once for the entire test class."""
+        # Create a single team which will cascade create division, season, competition, and club
+        cls.team = TeamFactory()
+        cls.division = cls.team.division
+        cls.season = cls.division.season
+        cls.competition = cls.season.competition
+        cls.club = cls.team.club
+        
+        # Create a simple PNG image
+        img = Image.new("RGB", (100, 100), color="red")
+        img_bytes = io.BytesIO()
+        img.save(img_bytes, format="PNG")
+        img_bytes.seek(0)
+        cls.png_data = img_bytes.read()
+        
+        # Create a simple JPEG image
+        img = Image.new("RGB", (100, 100), color="blue")
+        img_bytes = io.BytesIO()
+        img.save(img_bytes, format="JPEG")
+        img_bytes.seek(0)
+        cls.jpg_data = img_bytes.read()
+        
+        # Create a simple SVG (text-based)
+        cls.svg_data = b'<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100"></svg>'
+        
+        # Create invalid file (PDF)
+        cls.pdf_data = b"%PDF-1.4 fake pdf content"
+
+    def test_logo_fields_optional(self):
+        """Verify logo fields are optional (blank=True, null=True)."""
+        # ImageField.name is None when no file is uploaded
+        self.assertIsNone(self.competition.logo_colour.name)
+        self.assertIsNone(self.competition.logo_monochrome.name)
+
+    def test_competition_logo_via_admin(self):
+        """Test that Competition edit view includes logo fields."""
+        namespace = self.competition._get_admin_namespace()
+        args = self.competition._get_url_args()
+        
+        with self.login(self.superuser):
+            response = self.get(f"{namespace}:edit", *args)
+            self.response_200()
+            self.assertContains(response, "logo_colour")
+            self.assertContains(response, "logo_monochrome")
```

Comment:
Use `assertResponseContains` and it MUST be a valid HTML fragment; do not just test for a text content substring that won't cause a failure later if the markup is changed.